File: 1176.cpp
#include <stdio.h>
#define uint64 unsigned long long int
uint64 fib[61];
int last = 2;

uint64 fibo(int n) {
	int i = 0;
	if (n < last) return fib[n];
	for (i = last; i <= n; i++) {
		fib[i] = fib[i-2] + fib[i-1];
		last = i;
	}
	return fib[n];
}
int main() {
	
	int n, t, i, j;
	scanf("%d", &t);
	fib[0] = 0;
	fib[1] = 1;
	
	for (i = 0; i < t; i++) {
		scanf("%d", &n);
		fibo(n);
		printf("Fib(%d) = %llu\n", n, fib[n]);	
	}
	return 0;
}

File: 1176.pas
var
	fib: array[0..61] of uint64;
	n, t, i, j: int32;
begin
	n:= 0;
	t:= 0;
	fib[0] := 0;
	fib[1] := 1;
	readLn(t);
	For i := 1 to t do
	begin
		readLn(n);
		For j := 2 to n do
		begin
			fib[j] := fib[j - 2] + fib[j - 1];
		end;
		writeln('Fib(', n, ') = ', fib[n]);
	end;
	
	
end.
